profile_bluesky/startup/instrument/plans/energy_plans.py

```diff
@@ -32,12 +33,74 @@ def moveE(energy):
             args_list[0] += (undulator.downstream.start_button,1)
 
     @stage_decorator([mono,undulator.downstream])
-    def inner():
+    def _inner_moveE():
         for args in args_list:
-            yield from mv(*args)
+            yield from mv(*args,group=group)
+
+    return (yield from _inner_moveE())
+
+def Escan_list(detectors, energy_list, md = None):
+
+    _positioners= [mono.energy]
+    if undulator.downstream.tracking:
+        _positioners.append(undulator.downstream.energy)
+
+    _md = {'detectors': [det.name for det in detectors],
+           'positioners': [pos.name for pos in _positioners],
+           'num_points': len(energy_list),
+           'num_intervals': len(energy_list)- 1,
+           'plan_args': {'detectors': list(map(repr, detectors)),
+                         'energy_list': list(map(repr, energy_list))},
+           'plan_name': 'Escan_list',
+           'hints': {},
+           }
+
+    _md.update(md or {})
+
+    _md['hints'] = {'dimensions': [(['monochromator_energy'],'primary')]}
+    _md['hints'].update(md.get('hints', {}) or {})
+
+    #abs_plot = AbsorptionCallback()
+    #@subs_decorator(Absorption)
+    @run_decorator(md=_md)
+    def _inner_Escan_list():
+        for energy in energy_list:
+            grp = short_uid('set')
+            yield Msg('checkpoint')
+            yield from moveE(energy,group=grp)
+            yield from trigger_and_read(list(detectors)+_positioners)
+
+    return (yield from _inner_Escan_list())
+
+def Escan(detectors,energy_0, energy_f, steps, md = None):
+
+    _md = {'plan_args': {'detectors': list(map(repr, detectors)),
+                         'initial_energy': repr(energy_0),
+                         'final_energy': repr(energy_f),
+                         'steps': repr(steps)},
+           'plan_name': 'Escan',
+           'hints': {'x':['mono_energy']},
+           }
+
+    _md.update(md or {})
+    energy_list = linspace(energy_0,energy_f,steps)
+    return (yield from Escan_list(detectors,energy_list,md=_md))
+
+#TODO: not sure where to put this. Maybe it is best fit for the mono?
```

Comment:
If you believe the qxscan is buried in `energy_plans.py`, perhaps it belongs in a separate file.  Whatever makes it easier to understand where to find code.

Reply:
These `# TODO:` were left overs from moving `qxscan_setup` to a device (it was here before).